internal/framework/fetch/fetch_test.go

```diff
@@ -0,0 +1,720 @@
+package fetch
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+)
+
+func TestGetRemoteFile(t *testing.T) {
+	fetcher, err := NewDefaultFetcher()
+	if err != nil {
+		t.Fatalf("NewDefaultFetcher() failed: %v", err)
+	}
```

Comment:
We can still use the gomega library to call the matchers to check for success.

See some of our other test files for examples. (`g.Expect(err).ToNot(HaveOccurred())`)

Reply:
Probably a lot in this test file that can be updated to use the proper matchers.